core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -93,18 +93,107 @@ func generateFungibleAssetLockContractId(ctx contractapi.TransactionContextInter
 	return contractId
 }
 
+/*
+ * Function to validate the locker in asset agreement.
+ * If locker is not set, it will be set to the caller.
+ * If the locker is set already, it ensures that the locker is same as the creator of the transaction.
+ */
+func validateAndSetLockerOfAssetAgreement(ctx contractapi.TransactionContextInterface, assetAgreement *common.AssetExchangeAgreement) error {
+	txCreator, err := ctx.GetStub().GetCreator()
+	if err != nil {
+		return fmt.Errorf("unable to get the transaction creator information: %+v", err)
+	}
+	if len(assetAgreement.Locker) == 0 {
+		assetAgreement.Locker = string(txCreator)
+	} else if assetAgreement.Locker != string(txCreator) {
+		return fmt.Errorf("locker %s in the asset agreement is not same as the transaction creator %s", assetAgreement.Locker, txCreator)
+	}
+
+	return nil
+}
+
+/*
+ * Function to validate the locker in fungible asset agreement.
+ * If locker is not set, it will be set to the caller.
+ * If the locker is set already, it ensures that the locker is same as the creator of the transaction.
+ */
+func validateAndSetLockerOfFungibleAssetAgreement(ctx contractapi.TransactionContextInterface, assetAgreement *common.FungibleAssetExchangeAgreement) error {
+	txCreator, err := ctx.GetStub().GetCreator()
+	if err != nil {
+		return fmt.Errorf("unable to get the transaction creator information: %+v", err)
+	}
+	if len(assetAgreement.Locker) == 0 {
+		assetAgreement.Locker = string(txCreator)
+	} else if assetAgreement.Locker != string(txCreator) {
+		return fmt.Errorf("locker %s in the fungible asset agreement is not same as the transaction creator %s", assetAgreement.Locker, txCreator)
+	}
+
+	return nil
+}
+
+/*
+ * Function to validate the recipient in asset agreement.
+ * If recipient is not set, it will be set to the caller.
+ * If the recipeint is set already, it ensures that the recipient is same as the creator of the transaction.
+ */
+func validateAndSetRecipientOfAssetAgreement(ctx contractapi.TransactionContextInterface, assetAgreement *common.AssetExchangeAgreement) error {
```

Comment:
You can probably just pass `Recipient` instead of the whole agreement here, so this function can be called by the locking functions that rely on contract IDs.

Reply:
This is not possible since we are also setting `Recipient` in case it was not set earlier.